[CIAPP-2540] Add buildkite CI app integration docs (#11471)
DataDog/documentation: content/en/continuous_integration/setup_pipelines/buildkite.md

---
title: Set up Tracing on a Buildkite Pipeline
kind: documentation
further_reading:
    - link: "/continuous_integration/explore_pipelines"
      tag: "Documentation"
      text: "Explore Pipeline Execution Results and Performance"
    - link: "/continuous_integration/troubleshooting/"
      tag: "Documentation"
      text: "Troubleshooting CI"
---

{{< site-region region="us,eu" >}}
## Configuring the Datadog integration

> The Datadog intragration for Buildkite works by using [webhooks][1] to send pipeline data to Datadog.

Reviewer: ```suggestion
The Datadog integration for [Buildkite][2] works by using [webhooks][1] to send pipeline data to Datadog.
```

Go to **Settings > Notification Services** in Buildkite and add a new webhook:

{{< site-region region="us" >}}

> * **Webhook URL**: `https://webhooks-http-intake.logs.datadoghq.com/api/v2/webhook/?dd-api-key=<API_KEY>` where `<API_KEY>` is [your Datadog API key][1].

Reviewer 2: ```suggestion
* **Webhook URL**: `https://webhook-intake.datadoghq.com/api/v2/webhook/?dd-api-key=<API_KEY>` where `<API_KEY>` is [your Datadog API key][1].
```

* **Events**: Select `job.finished` and `build.finished`.
* **Pipelines**: Select all pipelines or the subset of pipelines you want to trace.

[1]: https://app.datadoghq.com/organization-settings/api-keys
{{< /site-region >}}

{{< site-region region="eu" >}}

> * **Webhook URL**: `https://webhooks-http-intake.logs.datadoghq.eu/api/v2/webhook/?dd-api-key=<API_KEY>` where `<API_KEY>` is [your Datadog API key][1].

Reviewer 2: ```suggestion
* **Webhook URL**: `https://webhook-intake.datadoghq.eu/api/v2/webhook/?dd-api-key=<API_KEY>` where `<API_KEY>` is [your Datadog API key][1].
```

[1]: https://app.datadoghq.eu/account/settings#api
{{< /site-region >}}

To set custom `env` or `service` parameters, add more query parameters in the webhooks URL: `&env=<YOUR_ENV>&service=<YOUR_SERVICE_NAME>`.

## Visualize pipeline data in Datadog

{{< site-region region="us" >}}
After the integration is successfully configured, the [Pipelines][1] and [Pipeline Executions][2] pages populate with data after the pipelines finish.

[1]: https://app.datadoghq.com/ci/pipelines
[2]: https://app.datadoghq.com/ci/pipeline-executions
{{< /site-region >}}

{{< site-region region="eu" >}}
After the integration is successfully configured, the [Pipelines][1] and [Pipeline Executions][2] pages populate with data after the pipelines finish.

[1]: https://app.datadoghq.eu/ci/pipelines
[2]: https://app.datadoghq.eu/ci/pipeline-executions
{{< /site-region >}}

**Note**: The Pipelines page shows data for only the default branch of each repository.

## Further reading

{{< partial name="whats-next/whats-next.html" >}}

{{< site-region region="us3,gov" >}}
The selected Datadog site ({{< region-param key="dd_site_name" >}}) is not supported at this time.

> {{< /site-region >}}

Reviewer: This should be outside of the last `{{< /site-region >}}`, otherwise it does not work.

> [1]: https://buildkite.com/docs/apis/webhooks

Reviewer: ```suggestion
[1]: https://buildkite.com/docs/apis/webhooks
[2]: https://buildkite.com
```

      text: "Explore Pipeline Execution Results and Performance"
    - link: "/continuous_integration/troubleshooting/"
      tag: "Documentation"
      text: "Troubleshooting CI"
---

{{< site-region region="us,eu" >}}
## Configuring the Datadog integration

The Datadog integration for [Buildkite][2] works by using [webhooks][1] to send pipeline data to Datadog.

Go to **Settings > Notification Services** in Buildkite and add a new webhook:

{{< site-region region="us" >}}
* **Webhook URL**: `https://webhooks-http-intake.logs.datadoghq.com/api/v2/webhook/?dd-api-key=<API_KEY>` where `<API_KEY>` is [your Datadog API key][1].
* **Events**: Select `job.finished` and `build.finished`.
* **Pipelines**: Select all pipelines or the subset of pipelines you want to trace.

[1]: https://app.datadoghq.com/account/settings#api
{{< /site-region >}}

{{< site-region region="eu" >}}
* **Webhook URL**: `https://webhooks-http-intake.logs.datadoghq.eu/api/v2/webhook/?dd-api-key=<API_KEY>` where `<API_KEY>` is [your Datadog API key][1].
* **Events**: Select `job.finished` and `build.finished`.
* **Pipelines**: Select all pipelines or the subset of pipelines you want to trace.

> [1]: https://app.datadoghq.eu/account/settings#api

Reviewer 3: Shouldn't this be https://app.datadoghq.com/organization-settings/api-keys ?

Author: You are right, I'll change it, but we also have to change it in the GitLab docs (I copied the URL from there), I'll open another ticket and update that one as well.

{{< /site-region >}}

> To set custom `env` or `service` parameters, add more query parameters in the webhooks URL: `&env=<YOUR_ENV>&service=<YOUR_SERVICE_NAME>`.

Reviewer 3: As a user, I wouldn't know what `env` and `service` mean in this context

Author: I'll add a link to https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging/?tab=kubernetes#overview which has the explanation of this reserved env vars.

Reviewer 3: I think we are using the `service` tag as the "CI provider instance id".

Author: Do we have that documented somewhere? It seems like it doesn't match the `service` in Data Dog as a whole which I assume would cause confusion for customer using multiple products.

Reviewer 3: I saw that mentioned here: https://docs-staging.datadoghq.com/carlos.gonzalez/add-buildkite-integration-docs/continuous_integration/setup_pipelines/gitlab/?tab=gitlabcom#configuring-the-datadog-integration

Reviewer 3: If buildkite doesn't support an on-prem version, it might just make sense to not document the `service` tag for buildkite.

Author: Is interesting because in that case does `env` have any value? We use it to separate between staging and prod and so on, but Buildkite only has one instance so both staging and prod would run from the same Buildkite thus `env` is always the same.

I guess you could setup multiple webhooks for different pipelines with different branch rules but that seems like a lot of work for the customer. You could do the same with services if you wanted to map some pipeline to some service namespace.

I find it a bit weird that our service does not match the general consensus of what service is in data dog.

Reviewer 2: These are for advanced use cases and we always default to empty. If there is no way to configure different instances of Buildkite I'd say they are not needed. 

The plan is to rename `service` to `ci.instance` this is why we have this facet already, but we never go to changing it in query params and docs.

Reviewer: Let's remove it from here. And also, we should remove them from the instructions of GitLab.com

Reviewer 2: why would we remove them from Gitlab?

Reviewer: From the SaaS section I meant
